src/Adapter/SDL/Render/Engine.php

```diff
@@ -122,7 +124,10 @@ public function start(Presentation\SlideShow $slideShow, Graphic\Drawer $drawer)
 
             // Screen refresh
             \SDL_RenderPresent($this->renderer);
-            \SDL_Delay(1000 / 60); /* 60 FPS */
+
+            // Frame rate capping
+            $elapsedTime = (int) (microtime(true) * 1000) - $iterationStartTime;
+            \SDL_Delay(max(0, 16 - $elapsedTime)); /* 60 FPS */
```

Comment:
What is the meaning of `16`?  I'm pretty sure that's the result of `1000/*ms*/ / 60/*frames*/` but… it's not really clear 😄 . What about letting `1000/*ms*/ / 60/*frames*/`? Thanks to Php7, the constant will be computed at compile time 🎉 

I think it's safer to ensure a minimum delay time… In case of heavy computing, it could be a good idea to let some time to the CPU… 

```suggestion
            \SDL_Delay(max(5, 1000/60 - $elapsedTime)); /* 1000ms / 60frames = time of 1 frame at 60 FPS */
```

Reply:
You're right, it's clearer to keep `1000/60` instead of `60`.

As for keeping a minimum 5ms pause, I disagree. The application goal is to render at 60 fps. If you set a minimum 5ms pause between each frame, it means that the application will run slower on some hardware even if it could run at an optimal speed (without this minimum pause).